### PR TITLE
record life cycle transitions is dynamodb

### PR DESCRIPTION
**Pull request by svozza**
*Description of changes:*

This records the lifecycle transitions in DynamoDb if the environment variable `PERSIST_LIFECYCLE_INFO` is set to `true`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/cfn/template.yaml [hunk 1]
```diff
@@ -1,105 +1,51 @@
-AWSTemplateFormatVersion: "2010-09-09"
```
Comment (matteofigus) [inline src/cfn/template.yaml]: This one looks like the autogenerated one in `templates/template.yaml`? I see some missing pieces (the vpc stuff), can you have a look?

File: src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        client.putItem(request);
+        logger.debug("Task " + status + " status written to DynamoDb...");
+    }
+
+    private Map<String, AttributeValue> toDynamoDbItem(String mediaUrl, String taskArn, String status) {
+        Map<String,AttributeValue> itemValues = new HashMap<>();
+
+        itemValues.put("MediaUrl", AttributeValue.builder().s(mediaUrl).build());
+        itemValues.put("TaskArn", AttributeValue.builder().s(taskArn).build());
+        itemValues.put("TaskStatus", AttributeValue.builder().s(status).build());
```
Comment (matteofigus) [inline src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java]: There should be more metadata here like `MediaDescription` and `MediaTitle`

File: src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    private String mediaUrl;
+
+    LifecycleInfoPersister(String mediaUrl) {
+        this.taskArn = Unirest
+            .get(TASK_METADATA_ENDPOINT)
+            .asJson()
+            .getBody()
+            .getObject()
+            .getString("TaskARN");
```
Comment (matteofigus) [inline src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java]: Should we also update the cfn template for changing the references from `TaskId` to `TaskARN`?

File: src/cfn/template.yaml (hunk 1 above)
Comment (svozza) [reply src/cfn/template.yaml]: ah shit, that must have happened when i did that weird rebase stuff. i'll revert.

File: src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java (hunk 3 above)
Comment (svozza) [reply src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java]: That was what my change was meant to do but it got lost when I did my rebase thing.

File: src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java (hunk 2 above)
Comment (svozza) [reply src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java]: Where do I get that info from?

File: src/cfn/template.yaml (hunk 1 above)
Comment (svozza) [reply src/cfn/template.yaml]: *done

Comment (svozza) [reply src/cfn/template.yaml]: hmmm, that didn't work

Comment (svozza) [reply src/cfn/template.yaml]: That's better.

File: src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java (hunk 2 above)
Comment (matteofigus) [reply src/backend/transcriber/src/main/java/com/amazonaws/transcriber/LifecycleInfoPersister.java]: As discussed, being this an upsert it's fine as is, perhaps `TaskArn` can also be removed as it's not being changed.

**Pull request merged**